js/index.js

```diff
@@ -1,42 +1,74 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = Number(product.querySelector('.price span').innerHTML);
+  const quantity = product.querySelector('.quantity input').valueAsNumber;
+  const total = price * quantity;
+  product.querySelector('.subtotal span').innerHTML = total.toFixed(2);
+  return total.toFixed(2);
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
   // ITERATION 2
-  //... your code goes here
+  const products = document.querySelectorAll('.product');
+  products.forEach((product) => {
+    updateSubtotal(product);
+  });
 
   // ITERATION 3
-  //... your code goes here
+  const subtotals = [];
+  products.forEach((product) => {
+    subtotals.push(Number(product.querySelector('.subtotal span').innerHTML));
+  });
+  const total = subtotals.reduce((x, y) => {
+    if (!subtotals[0]) return 0;
+    return x + y;
+  });
+  document.querySelector('#total-value span').innerText = total.toFixed(2);
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  const target = event.currentTarget.parentNode.parentNode;
+  document.getElementById('body').removeChild(target);
 }
 
 // ITERATION 5
 
 function createProduct() {
-  //... your code goes here
+  const table = document.getElementById('body');
+  const product = document.querySelector('.product').innerHTML;
+  const newProduct = document.createElement('tr');
+
+  newProduct.className = 'product';
+  newProduct.innerHTML = product;
+
+  const newProductName = document.querySelector('#name-input').value;
+  const newProductPrice = Number(
+    document.querySelector('#quantity-input').value
+  ).toFixed(2);
+
+  //this just seems logical... 😄
+  if (newProductName === '' || newProductPrice == 0) return;
+
+  newProduct.querySelector('.product span').innerText = newProductName;
+  newProduct.querySelector('.price span').innerText = newProductPrice;
+
```

Comment:
Here you need to change subtotal value to 0 ;) because when you create new product, it receive subtotal of first product ;)


`  newProduct.querySelector('.subtotal span').innerText= 0
`